Dart/src/com/jetbrains/lang/dart/ide/actions/DartPubActionBase.java

```diff
@@ -59,11 +60,23 @@
 import static com.jetbrains.lang.dart.util.PubspecYamlUtil.PUBSPEC_YAML;
 
 abstract public class DartPubActionBase extends AnAction implements DumbAware {
+  public static final String PUB_ENV_VAR_NAME = "PUB_ENVIRONMENT";
+
   private static final String GROUP_DISPLAY_ID = "Dart Pub Tool";
   private static final Key<PubToolWindowContentInfo> PUB_TOOL_WINDOW_CONTENT_INFO_KEY = Key.create("PUB_TOOL_WINDOW_CONTENT_INFO_KEY");
 
   private static final AtomicBoolean ourInProgress = new AtomicBoolean(false);
 
+  public static String getPubEnvValue() {
+    String existingVar = System.getenv(PUB_ENV_VAR_NAME);
+    if (existingVar == null) {
+      return DartAnalysisServerService.getClientId();
+    }
+    else {
+      return existingVar + ":" + DartAnalysisServerService.getClientId();
```

Comment:
I don't think so; this is to create a new, longer env var token, so that if we're running in another context - like on a travis bot - we get a different token. It could then we used to differentiate between user usage and bot usage.

(so the separator token doesn't have much meaning here, just something to concatenate with)